### DISCUSSION TITLE
A way to exit `lfcd` without changing the directory

### DISCUSSION THREAD

**Discussion by ilyagr** [gokcehan/lf#915] [Q&A] [answered]
I would like `q` to exist [`lfcd`](https://github.com/gokcehan/lf/blob/master/etc/lfcd.fish) without changing the directory, and `Q` to exit and cd to the current directory. Is there a way of achieving this?

If not, I'd appreciate this as a feature (command to exit `lf` without respecting the `-last-dir-to-string` option). *Update*: Or, perhaps, just a command to quit with an exit code would be sufficient.

Another (inferior, IMO) alternative would be to remember the directory `lf` started in. Then, I could do something like `map q cd_starting_dir; quit`. I looked for such a thing, but missed it if it's there.

**Update:** The [`ranger` analogue](https://github.com/ranger/ranger/blob/master/examples/fish_automatic_cd.fish) suggests using `cd -` after existing, which is a good point but not ideal.

**Comment by ilyagr** [comment, marked as answer]
After some adventures with quoting and irritation that `cd $OLDPWD` doesn't work,
I think I found a way that does work:

```
map q %{{
  lf -remote "send $id :cd $OLDPWD; quit"
}}
map Q quit
```

**Comment by DusanLesan** [comment]
Try:
```
push m-
map q push '-:quit<enter>
map Q quit
```
From [Issue 140](https://github.com/gokcehan/lf/issues/140)

**Reply by ilyagr** [reply]
This works great, thanks!

**Reply by ilyagr** [reply]
**Update**: This comment is outdated now.

Actually, correction, this doesn't work great because the mapping is cyclic. It's easy to fix, though:

```
push m-
map q push '-:quit<enter>
map Q quit
```

Perhaps you should edit your answer and then I can accept it again?

**Reply by DusanLesan** [reply]
The code is changed to the one you mentioned. As I have remapped from `m` to `~` to use m with some other combinations, I use:
```
push ~w
map Q push 'w:quit<enter>
```

**Reply by ilyagr** [reply]
Actually, I found a (different) problem with this entire approach. It doesn't work if you start lf as `lf some/path`. See also https://github.com/gokcehan/lf/issues/140#issuecomment-1229128391.

**Reply by ilyagr** [reply]
I think I solved the puzzle. See my other answer.

**Reply by DusanLesan** [reply]
Nice, I was trying some similar approaches at first, but I was never quite there, I had some messy version where I wrote a file in tempfs that was validated after exit in lfcd.

**Reply by CSWLeTianPai** [reply]
how to make it run in windows?
